Firebase/Storage/FIRStorageUploadTask.m

```diff
@@ -177,6 +181,37 @@ - (void)enqueue {
   }];
 }
 
+- (void)finishTaskWithStatus:(FIRStorageTaskStatus)status
+                    snapshot:(FIRStorageTaskSnapshot *)snapshot {
+  [self fireHandlersForStatus:status snapshot:self.snapshot];
+  [self removeAllObservers];
+  self->_fetcherCompletion = nil;
+}
+
+- (BOOL)isContentToUploadValid:(NSError **)outError {
+  if (_uploadData != nil) {
+    return YES;
+  }
+
+  NSError *fileReachabilityError;
+  if (![_fileURL checkResourceIsReachableAndReturnError:&fileReachabilityError]) {
+    if (outError != NULL) {
+      NSString *description =
+        [NSString stringWithFormat:@"File at URL: %@ is not reachable", _fileURL.absoluteString];
+      *outError = [NSError errorWithDomain:FIRStorageErrorDomain
+                                      code:FIRStorageErrorCodeUnknown
```

Comment:
I think this is fine for now, but we probably want to add a case to the to the `FIRStorageErrorCode` enum.

That will require an API review (I can go over the process with you) so we won't be able to do it this release.

Reply:
I may create an issue for that so we don't forget about it.

Reply:
Once this is approved by the Storage team and lands, that's a great idea.

Reply:
Created - #2459

Reply:
Just throwing this out there: Did you consider FIRStorageErrorCodeObjectNotFound? It's used today when a file doesn't exist on the backend, which isn't all that different from a locally missing file.

Reply:
My concern with `FIRStorageErrorCodeObjectNotFound` is following: a typical file upload operation involves a remote object reference and a file URL, so if we have a single error code for both it may be confusing for users.